ConfigurationSystem/Client/PathFinder.py

```diff
@@ -30,29 +30,45 @@ def getSystemSection( serviceName, serviceTuple = False, instance = False, setup
     instance = getSystemInstance( serviceTuple[0], setup = setup )
   return "/Systems/%s/%s" % ( serviceTuple[0], instance )
 
+def getComponentSection (componentName, componentTuple = False, setup = False, componentCategory ="Services"):
+  """Function returns the path to the component.
+
+  Args:
+    componentName(str): Component name prefixed by the system in which it is placed.
+                        e.g. 'WorkloadManagement/SandboxStoreHandler'
+    componentTuple
+    setup(str): Name of the setup.
+    componentCategory(str): Category of the component, it can be: 'Agents', 'Services', 'Executors', 'Consumers'
+                            or 'Databases'.
+
+  Returns:
+    str: Complete path to the component
+
+  Raises:
+    RuntimeException: If in the componentName - the system part does not correspond to any known system in DIRAC.
+
+  Example:
+    getComponentSection('WorkloadManagement/SandboxStoreHandler', False,False,'Services')
+  """
+  if not componentTuple:
+    componentTuple = divideFullName( componentName )
+  systemSection = getSystemSection( componentName, componentTuple, setup = setup )
+  return "%s/%s/%s" % ( systemSection,componentCategory,  componentTuple[1] )
```

Comment:
Ok, I agree, however since I was just refactoring/generalizign the existing functions I used the type that was already used there. I can change it, but that would also mean , to trace back all the calls of getServiceSection, getAgentSection and so on  and check what is going on there.
By the way, the situations where the error appear are completely non-intiutive - one can see it in unit tests. I left it cause again, that would probably mean changes in more sections of code.